compiler/util/misc.cpp

```diff
@@ -485,6 +485,31 @@ static void printCallstackForLastError() {
   }
 }
 
+//
+// Given a function whose name starts with chpl_, it may be of interest
+// to a non-developer if it is a "non-trivial" module init function.
+// As a heuristic, we consider a module to be "trivial" if its name matches
+// the name of its file or if it starts on the first line of its file.
+//
+static bool nontrivialModuleInit(FnSymbol* fn) {
+  if (! fn->hasFlag(FLAG_MODULE_INIT))
+    return false;
+
+  if (fn->linenum() <= 1)
+    return false;
+
+  const char* basename = fn->fname();
+  if (basename == nullptr)
+    return false;
+
+  if (const char* slash = (const char*) strrchr(basename, '/'))
+    basename = slash+1;
+
+  const char* modulename = astr(fn->name + 11, ".chpl");
```

Comment:
this seems unsafe to me because we don't know the length of fn->name. Sure, something with the flag checked for above should start with `chpl__init_` (or something along those lines) but maybe we need to have an assertion here checking it starts with what we expect. You can use `startsWith` to check.